# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStore.cpp
# Commit: 16c36964a84b
# Full Hash: 16c36964a84ba18209a6a86b59b5e5dc20e60925
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-12-03 21:40:41
# Description:
#   Bug 1934330 - CookieStore - check the existence of the window, r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D230920
# ==============================================================================

diff -r d140333670bc -r 16c36964a84b dom/cookiestore/CookieStore.cpp
--- a/dom/cookiestore/CookieStore.cpp	Tue Dec 03 15:55:00 2024 +0200
+++ b/dom/cookiestore/CookieStore.cpp	Tue Dec 03 13:50:03 2024 +0000
@@ -638,7 +638,10 @@
 
           if (NS_IsMainThread()) {
             nsCOMPtr<nsPIDOMWindowInner> window = self->GetOwnerWindow();
-            MOZ_ASSERT(window);
+            if (NS_WARN_IF(!window)) {
+              promise->MaybeReject(NS_ERROR_DOM_SECURITY_ERR);
+              return;
+            }
 
             nsCOMPtr<Document> document = window->GetExtantDoc();
             if (NS_WARN_IF(!document)) {
